### PR TITLE
OKE-26958 Web application firewall integration

### PR DESCRIPTION
**Pull request by Inbaraj-S**
- Added support for annotation `oci-native-ingress.oraclecloud.com/waf-policy-ocid` which enables user to add firewall policy id to IngressClass. It helps us create and add firewall to the corresponding LB.
- Post creation of firewall, we add the corresponding firewall ocid to the ingressclass via `oci-native-ingress.oraclecloud.com/firewall-id`.
- [OKE-27261-Fixed-multi-path-bug](https://github.com/oracle/oci-native-ingress-controller/pull/14/commits/b90ba24f5e3313857c6d7ca999b31a97aff7db01)

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package client
+
+import (
+	"github.com/oracle/oci-native-ingress-controller/pkg/certificate"
+	"github.com/oracle/oci-native-ingress-controller/pkg/loadbalancer"
+	"github.com/oracle/oci-native-ingress-controller/pkg/waf"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Client struct {
```
Comment (nirpai) [inline pkg/client/client.go:10]: Can we modify the name of this struct to ClientProvider?

File: pkg/controllers/ingressclass/ingressclass.go [hunk 2]
```diff
@@ -283,25 +279,36 @@ func (c *Controller) ensureLoadBalancer(ic *networkingv1.IngressClass) error {
...
 			return err
 		}
 	}
 
 	klog.V(4).InfoS("checking if updates are required for load balancer", "ingressClass", klog.KObj(ic))
+	return nil
+}
 
+func (c *Controller) setupWebApplicationFirewall(ic *networkingv1.IngressClass, compartmentId *string, lbId *string) error {
+	firewall, err, err2, done := c.wafClient.GetFireWallId(c.client, ic, compartmentId, lbId)
```
Comment (nirpai) [inline pkg/controllers/ingressclass/ingressclass.go]: Can we add a comment on what would be value of done here and on what case we return err2? (or update variables accordingly?)

**Review by nirpai — APPROVED** [review]

File: pkg/client/client.go (hunk 1 above)
Comment (Inbaraj-S) [reply pkg/client/client.go:10]: Done

File: pkg/controllers/ingressclass/ingressclass.go (hunk 2 above)
Comment (Inbaraj-S) [reply pkg/controllers/ingressclass/ingressclass.go]: Done

**Pull request merged**